zerver/lib/message.py

```diff
@@ -1403,6 +1411,10 @@ def format_unread_message_details(
         )
         if message_id in raw_unread_data["mentions"]:
             message_details["mentioned"] = True
+            if message_id in raw_unread_data["mentions_me_directly"]:
+                message_details["mentioned_me_directly"] = True
+            elif message_id in raw_unread_data["wildcard_mentions"]:
```

Comment:
you might want to make these regular `if` statements since someone could probably make a message that contains both a direct mention of a user as well as a wildcard mention, and this `elif` would never properly catch those cases